go/tasks/plugins/k8s/kfoperators/pytorch/pytorch.go

```diff
@@ -68,20 +68,35 @@ func (pytorchOperatorResourceHandler) BuildResource(ctx context.Context, taskCtx
 
 	common.OverrideDefaultContainerName(taskCtx, podSpec, kubeflowv1.PytorchJobDefaultContainerName)
```

Comment:
@hamersaw As discussed I now first override the primary container name with the pytorch job default container name. Next, the podSpecs are merged, specifying the pytorch job default container name as the primary container name to the `MergePodSpecs` function.